imports/plugins/core/orders/client/components/orderSearch.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from "react";
+import { compose } from "recompose";
+import PropTypes from "prop-types";
+import { Components, registerComponent, withPermissions } from "@reactioncommerce/reaction-components";
+
+/**
+ * React class for Search bar on Order Dashboard
+ * @summary horizontal search bar on the order dashboard. can be replaced with registerComponent as "OrderSearch"
+ * @property {Function} handleChange - function called to update state field on parent after search input text changes
+ */
+class OrderSearch extends Component {
+  static propTypes = {
+    handleChange: PropTypes.func
+  };
+
+  constructor() {
+    super();
+
+    this.state = { query: "" };
+  }
+
+  /**
+   * handleChange - handler to call onchange of search input
+   * @param {string} event - event object
+   * @param {string} value - current value of the search input
+   * @return {null} -
+   */
+  handleChange = (event, value) => {
+    this.setState({ query: value });
+
+    if (this.props.handleChange) {
```

Comment:
We should probably remove this conditional and make this property `isRequired`. There's no good reason to never handle this handler. If it's not there we should probably throw an error rather than just do nothing?